packages/studio-plugin/src/ParsingOrchestrator.ts

```diff
@@ -135,12 +149,10 @@ export default class ParsingOrchestrator {
       }
     } else if (filepath.startsWith(this.paths.layouts)) {
       const layoutName = upath.basename(filepath, ".tsx");
-      const index = this.layoutNames.indexOf(layoutName);
-      if (index >= 0) {
-        this.layoutNames.splice(index, 1);
-      }
+      delete this.layoutNameToLayoutFile[layoutName];
       if (fileExists) {
-        this.layoutNames.push(layoutName);
+        this.layoutNameToLayoutFile[layoutName] =
+          this.getLayoutFile(layoutName);
```

Comment:
The way we handle `Layouts` and `Pages` in `ParsingOrchestrator` is so similar. I wonder if we could figure out a nice way to have consolidated, common logic that both use.

Reply:
Consolidated most of the logic into generic functions that are used by both. Some of the module stuff could also have been consolidated, but I left that alone to avoid merge conflicts